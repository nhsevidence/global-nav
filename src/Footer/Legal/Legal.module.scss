@import '~@nice-digital/nds-core/scss/core';
@import '../footer-settings';

.wrapper {
  @include nds-font-size(-1);
  background: $nds-colour-nice-dark-grey;
  padding: rem($nds-spacing-medium 0);

  a {
    text-decoration: underline;
  }
}

.container {
  @include nds-container;
}

.menu {
  @include nds-hide-print;
  margin: rem(0 0 $nds-spacing-medium);

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  li {
    margin: 0;
  }

  a {
    display: block;
  }
}

.copyright {
  @include nds-font-size(-1);
  margin: 0;
  @media print {
    float: none !important;
    max-width: 100%;
  }
}

@include mq($from: sm) {
  .menu {
    li {
      display: inline-block;
      margin-right: rem($nds-spacing-medium);
    }
  }
}

@include mq($from: lg) {
  .wrapper {
    @include nds-clearfix;
  }

  .menu {
    float: left;
    margin: 0;
  }

  .copyright {
    float: right;
  }
}
