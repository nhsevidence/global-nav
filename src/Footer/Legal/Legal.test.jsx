import React from "react";
import MockDate from "mockdate";

import Legal from "./Legal";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Legal", () => {
	it("Renders without crashing", () => {
		const wrapper = shallow(<Legal />);
		expect(wrapper).toHaveLength(1);
	});

	it("Matches snapshot", () => {
		MockDate.set("2000-11-22");
		const wrapper = shallow(<Legal />);
		expect(toJson(wrapper)).toMatchSnapshot();
		MockDate.reset();
	});
});
