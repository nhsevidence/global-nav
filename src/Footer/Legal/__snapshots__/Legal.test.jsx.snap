// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legal Matches snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="container"
  >
    <nav
      aria-label="Legal menu"
      className="menu"
    >
      <ul>
        <li>
          <TrackedLink
            eventAction="Footer click"
            eventCategory="TopHat and footer"
            eventLabel="Accessibility"
            href="https://www.nice.org.uk/accessibility"
          >
            Accessibility
          </TrackedLink>
        </li>
        <li>
          <TrackedLink
            eventAction="Footer click"
            eventCategory="TopHat and footer"
            eventLabel="Freedom of information"
            href="https://www.nice.org.uk/freedom-of-information"
          >
            Freedom of information
          </TrackedLink>
        </li>
        <li>
          <TrackedLink
            eventAction="Footer click"
            eventCategory="TopHat and footer"
            eventLabel="Glossary"
            href="https://www.nice.org.uk/glossary"
          >
            Glossary
          </TrackedLink>
        </li>
        <li>
          <TrackedLink
            eventAction="Footer click"
            eventCategory="TopHat and footer"
            eventLabel="Terms and conditions"
            href="https://www.nice.org.uk/terms-and-conditions"
          >
            Terms and conditions
          </TrackedLink>
        </li>
        <li>
          <TrackedLink
            eventAction="Footer click"
            eventCategory="TopHat and footer"
            eventLabel="Privacy notice"
            href="https://www.nice.org.uk/privacy-notice"
          >
            Privacy notice
          </TrackedLink>
        </li>
      </ul>
    </nav>
    <p
      className="copyright"
    >
      Â©
       
      <abbr
        title="National Institute for Health and Care Excellence"
      >
        NICE
      </abbr>
       
      2000
      . All rights reserved. Subject to
       
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Notice of rights"
        href="https://www.nice.org.uk/terms-and-conditions#notice-of-rights"
      >
        Notice of rights
      </TrackedLink>
      .
    </p>
  </div>
</div>
`;
